Userland/Libraries/LibJS/Bytecode/Pass/LoadElimination.cpp

```diff
@@ -0,0 +1,152 @@
+
+                    if (array_ranges.get(store.dst().index())) {
+                        // The register is part of an array, so the store to it cannot
+                        // be simply replaced by a reference to another register
+                        new (new_block->next_slot()) Op::Store(store);
+                        new_block->grow(sizeof(Op::Store));
+                    } else {
+                        register_table.set(store.dst().index(), reg->value);
+                    }
+
+                    ++it;
+                }
+
+                continue;
+            }
+
+            new (new_block->next_slot()) Op::GetVariable(get_variable);
+            new_block->grow(sizeof(Op::GetVariable));
+
+            if (next_instruction.type() == Instruction::Type::Store) {
+                auto const& store = static_cast<Op::Store const&>(next_instruction);
+                identifier_table.set(get_variable.identifier().value(), store.dst());
+
+                new (new_block->next_slot()) Op::Store(store);
+                new_block->grow(sizeof(Op::Store));
+                ++it;
+            }
+
+            continue;
+        }
+        case SetVariable: {
+            auto const& set_variable = static_cast<Op::SetVariable const&>(*it);
+
+            identifier_table.remove(set_variable.identifier().value());
+
+            break;
+        }
+        case Store: {
+            auto const& store = static_cast<Op::Store const&>(*it);
+            // FIXME: This may be redundant due to being SSA like in our register usage
+            register_table.remove(store.dst().index());
+
+            break;
+        }
+        case Call:
+        case GetByValue:
+        case GetById:
+        case PutByValue:
+        case PutById:
+            // Note: Value getters/setters and calls, especially eval, can change variables as side effects
+            // FIXME: In strict mode and with better identifier metrics, we might be able
+            //        to safe some caching with a more fine-grained identifier table
+            memcpy(new_block->next_slot(), &*it, (*it).length());
+            for (auto regs : register_table)
+                reinterpret_cast<Instruction*>(new_block->next_slot())->replace_references(Register { regs.key }, regs.value);
+            new_block->grow((*it).length());
+
+            identifier_table.clear_with_capacity();
+
+            ++it;
+            continue;
+        case NewBigInt:
+            // FIXME: This is the only non trivially copyable Instruction,
+            //        so we need to do some extra work here
+            new (new_block->next_slot()) Op::NewBigInt(static_cast<Op::NewBigInt const&>(*it));
+            new_block->grow(sizeof(Op::NewBigInt));
+            ++it;
+            continue;
+        default:
+            break;
+        }
+
+        memcpy(new_block->next_slot(), &*it, (*it).length());
+        for (auto regs : register_table)
```

Comment:
That doesn't look like a "regs", it's just a single register

Reply:
chose a hopefully better name